File: app/src/main/java/cz/civilizacehra/cipherbreaker/StringPair.java
package cz.civilizacehra.cipherbreaker;

class StringPair {
    private String first;
    private String second;

    private StringPair(String first, String second) {
        super();
        this.first = first;
        this.second = second;
    }

    public boolean equals(Object other) {
        if (other instanceof StringPair) {
            StringPair otherPair = (StringPair) other;
            return
                    this.first != null && otherPair.first != null && this.first.equals(otherPair.first) &&
                            this.second != null && otherPair.second != null && this.second.equals(otherPair.second);
        }

        return false;
    }

    public String toString()
    {
        return "(" + first + ", " + second + ")";
    }

    static StringPair fromString(String input)
    {
        int index = input.indexOf(":");
        if (index > 0 && index < input.length()) {
            return new StringPair(input.substring(0, index), input.substring(index + 1, input.length()));
        } else {
            return new StringPair(input, input);
        }
    }

    String getFirst() {
        return first;
    }

    public void setFirst(String first) {
        this.first = first;
    }

    String getSecond() {
        return second;
    }

    public void setSecond(String second) {
        this.second = second;
    }
}

File: app/src/main/java/cz/civilizacehra/cipherbreaker/StringPair.kt
package cz.civilizacehra.cipherbreaker

internal class StringPair private constructor(val first: String?, val second: String?) {

    override fun equals(other: Any?): Boolean {
        if (other is StringPair) {
            return this.first != null && other.first != null && this.first == other.first &&
                    this.second != null && other.second != null && this.second == other.second
        }

        return false
    }

    override fun hashCode(): Int {
        return this.toString().hashCode()
    }

    override fun toString(): String {
        return "($first, $second)"
    }

    companion object {

        fun fromString(input: String): StringPair {
            val index = input.indexOf(":")
            return if (index > 0 && index < input.length) {
                StringPair(input.substring(0, index), input.substring(index + 1))
            } else {
                StringPair(input, input)
            }
        }
    }
}
